translator/translate/otel/pipeline/prometheus/translator.go

```diff
@@ -12,41 +13,98 @@ import (
+	dataType component.DataType
+	common.DestinationProvider
+}
+
+type Option func(any)
+
+func WithDataType(dataType component.DataType) common.TranslatorOption {
+	return func(a any) {
+		if t, ok := a.(*translator); ok {
+			t.dataType = dataType
+		}
+	}
 }
 
 var _ common.Translator[*common.ComponentTranslators] = (*translator)(nil)
 
-func NewTranslator() common.Translator[*common.ComponentTranslators] {
-	return &translator{}
+func NewTranslator(opts ...common.TranslatorOption) common.Translator[*common.ComponentTranslators] {
+	t := &translator{name: common.PipelineNamePrometheus}
+	for _, opt := range opts {
+		opt(t)
+	}
+	if t.Destination() != "" {
+		t.name += "/" + t.Destination()
+	}
+	return t
 }
 
 func (t *translator) ID() component.ID {
-	return component.NewIDWithName(component.DataTypeMetrics, pipelineName)
+	return component.NewIDWithName(component.DataTypeMetrics, t.name)
 }
 
 // Translate creates a pipeline for prometheus if the logs.metrics_collected.prometheus
 // section is present.
 func (t *translator) Translate(conf *confmap.Conf) (*common.ComponentTranslators, error) {
-	key := common.ConfigKey(common.LogsKey, common.MetricsCollectedKey, common.PrometheusKey)
-	if conf == nil || !conf.IsSet(key) {
-		return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: key}
+	if conf == nil || !conf.IsSet(common.PrometheusConfigKeys[t.dataType]) {
+		return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: fmt.Sprint(common.PrometheusConfigKeys[t.dataType])}
+	}
+
+	if t.dataType == component.DataTypeMetrics && !conf.IsSet(common.ConfigKey(common.PrometheusConfigKeys[t.dataType], common.PrometheusConfigPathKey)) {
+		return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: fmt.Sprint(common.ConfigKey(common.PrometheusConfigKeys[t.dataType], common.PrometheusConfigPathKey))}
+	}
+	// return pipeline based on destination to keep source/destination combinations clearly separated
+	// telegraf_prometheus - cloudwatch
+	// otel_prometheus - AMP
+	// this could change in future releases to support different source/destination combinations
+	switch t.Destination() {
+	case common.DefaultDestination, common.CloudWatchKey:
+		if !conf.IsSet(common.PrometheusConfigKeys[component.DataTypeLogs]) {
+			return nil, fmt.Errorf("pipeline (%s) is missing prometheus configuration under logs section with destination (%s)", t.name, t.Destination())
+		}
+		if !conf.IsSet(common.MetricsDestinations) || conf.IsSet(common.ConfigKey(common.MetricsDestinations, common.CloudWatchKey)) {
+			return &common.ComponentTranslators{
+				Receivers: common.NewTranslatorMap(adapter.NewTranslator(prometheus.SectionKey, common.PrometheusConfigKeys[t.dataType], time.Minute)),
+				Processors: common.NewTranslatorMap(
+					batchprocessor.NewTranslatorWithNameAndSection(t.name, common.LogsKey), // prometheus sits under metrics_collected in "logs"
+				),
+				Exporters:  common.NewTranslatorMap(awsemf.NewTranslatorWithName(common.PipelineNamePrometheus)),
```

Comment:
If the destination is "cloudwatch" from the metrics section, then the exporter should be the cloudwatchexporter (PMD).